src/Task.js

```diff
@@ -0,0 +1,50 @@
+import { isUndefined, defer, clearDefer } from './utils';
+
+/**
+ * A utility class to manage a single defferable task.
+ *
+ * @class Task
+ */
+
+export default class Task {
+  _id;
+
+  /**
+   * Checks if a task is set to execute in the future.
+   *
+   * @returns {boolean} True if a task is set.
+   */
+
+  isSet() {
+    return !isUndefined(this._id);
+  }
+
+  /**
+   * Defers a function to be executed on the next turn of the event loop.
+   * Will not allow a defer if the task is already set to execute.
+   *
+   * @param {Function} Callback to execute.
+   */
+
+  defer(callback) {
```

Comment:
For deferring? A promise represents the completion of some task. I think things like deferring and `setTimeout`s make more sense as a callback. For one they should never error. In this case a callback will also be slightly more performant.